From 9ef037bd1f14da16b3529f227fdf1e3307b9022d Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Thu, 11 Jun 2015 14:40:17 +0800
Subject: [PATCH 0603/5242] MLK-11093-1 dts: imx6ul-14x14-ddr-arm2: add SD1
 eMMC support

commit  31657f1201790448d21a315b41cb5b850a0e5c7e from
https://source.codeaurora.org/external/imx/linux-imx.git

The eMMC signals are reused from SD1 slot signals on MX6UL ARM2 board.
Thus a new dts is introduced to avoid such pin confliction.
Before using this dts, user has to do hw rework to connect eMMC IO
signals and disconnect SD1 Slot signals.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit e46408ad15bed829e6f254ced16524fcfe99176f)

Conflicts:
	arch/arm/boot/dts/Makefile
	arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2.dts
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                        |    1 +
 arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts |   20 ++++++++++++++++++++
 2 files changed, 21 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 924e1b8..3e249ee 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -568,6 +568,7 @@ dtb-$(CONFIG_SOC_IMX6UL) += \
 	imx6ul-tx6ul-0011.dtb \
 	imx6ul-tx6ul-mainboard.dtb \
 	imx6ul-14x14-ddr3-arm2.dtb \
+	imx6ul-14x14-ddr3-arm2-emmc.dtb	\
 	imx6ul-14x14-ddr3-arm2-gpmi-weim.dtb \
 	imx6ul-14x14-ddr3-arm2-mqs.dtb \
 	imx6ul-14x14-ddr3-arm2-spdif.dtb \
diff --git a/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts
new file mode 100644
index 0000000..234d3ce
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-14x14-ddr3-arm2-emmc.dts
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ul-14x14-ddr3-arm2.dts"
+
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1_8bit>;
+	bus-width = <8>;
+	cd-gpios = <>;
+	wp-gpios = <>;
+	vmmc-supply = <>;
+	non-removable;
+	status = "okay";
+};
-- 
1.7.9.5

